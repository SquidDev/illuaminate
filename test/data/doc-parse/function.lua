--- This is a short description
-- @param a Has no type
-- @tparam number b Has a type
-- @param[2] c A second argument set
-- @return Some thing
-- @treturn number A number
-- @return[2] Another return set
-- @treturn[3] number... Lots of me
-- @throws A problem

-- TODO: @param[type=number] foo A foo argument
