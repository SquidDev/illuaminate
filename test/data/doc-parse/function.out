((source function.lua[1:1-9:20])
 (description "This is a short description\n\n")
 (args ((name c) (description "A second argument set\n\n")))
 (args ((name a) (description "Has no type\n\n"))
  ((name b) (type number) (description "Has a type\n\n")))
 (returns ((type number) (many) (description "Lots of me\n\n")))
 (returns ((description "Another return set\n\n")))
 (returns ((description "Some thing\n\n"))
  ((type number) (description "A number\n\n"))) (throws "A problem\n\n"))
